File: components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/persistence/store/impl/Attribute.java
package org.wso2.carbon.status.dashboard.core.persistence.store.impl;

/**
 * Created by yasara on 9/13/17.
 */
public class Attribute {
    String name;
    String type;

    public Attribute(String name, String type) {
        this.name = name;
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }
}

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/elements/attribute.js
/**
 * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org)  Apache License, Version 2.0  http://www.apache.org/licenses/LICENSE-2.0
 */
define(
    function () {

        /**
         * @class Attribute
         * @constructor
         * @class Attribute  Creates an object to hold an attribute
         * @param {Object} options Rendering options for the view
         */
        var Attribute = function (options) {
            /*
             Data storing structure as follows
             {
             name: ‘’,
             type: ‘’
             }
             */
            if (options !== undefined) {
                this.name = options.name;
                this.type = (options.type !== undefined) ? (options.type).toUpperCase() : undefined;
            }
        };

        Attribute.prototype.getName = function () {
            return this.name;
        };

        Attribute.prototype.getType = function () {
            return this.type;
        };

        Attribute.prototype.setName = function (name) {
            this.define = name;
        };

        Attribute.prototype.setType = function (type) {
            this.type = type.toUpperCase();
        };

        return Attribute;

    });
